src/dbnode/encoding/m3tsz/encoder.go

```diff
@@ -276,8 +277,11 @@ func (enc *encoder) reset(start time.Time, bytes checked.Bytes) {
 }
 
 // Stream returns a copy of the underlying data stream.
-func (enc *encoder) Stream(opts encoding.StreamOptions) (xio.SegmentReader, bool) {
-	segment := enc.segment(byCopyResultType)
+func (enc *encoder) Stream(
+	ctx context.Context,
```

Comment:
Take it or leave it, but it could be nice to make the `zeroCopy` behavior optional via the `encoding.StreamOptions` and put the context in their too so that we don't have to pollute this API (and create a context everytime we want to do this even when we dont need zero copy)